torchci/lib/bot/autoLabelBot.ts

```diff
@@ -249,8 +314,55 @@ function myBot(app: Probot): void {
         break;
     }
 
-    if (newLabels.length) {
-      await addLabels(context, newLabels);
+    const newLabelsFromLabelToLabelConfig =
+      await getLabelsFromLabelToLabelConfig(
+        context,
+        labelToLabelConfigTracker,
+        labels,
+        label
+      );
+    for (const label of newLabelsFromLabelToLabelConfig) {
+      addLabel(labelSet, newLabels, label);
+    }
+
+    const filtered = await filterCIFlowLabels(true, newLabels);
+    if (filtered.length) {
+      await addLabels(context, filtered);
+    }
+  });
+
+  app.on("pull_request.labeled", async (context) => {
+    // Careful! Read the above comments about label addition
+
+    const label = context.payload.label!.name;
+    const labels: string[] = context.payload.pull_request.labels!.map(
+      (e) => e["name"]
+    );
+    context.log({ label, labels });
+
+    const labelSet = new Set(labels);
+    const newLabels: string[] = [];
+
+    const newLabelsFromLabelToLabelConfig =
+      await getLabelsFromLabelToLabelConfig(
+        context,
+        labelToLabelConfigTracker,
+        labels,
+        label
+      );
+    for (const label of newLabelsFromLabelToLabelConfig) {
+      addLabel(labelSet, newLabels, label);
+    }
+
+    const filtered = await filterCIFlowLabels(
+      false,
+      newLabels,
+      context,
+      context.payload.repository.owner.login,
+      context.payload.repository.name
+    );
+    if (filtered.length) {
+      await addLabels(context, filtered);
```

Comment:
Just curious, do we want to have a check here to see if filtered is the same as the list of existing labels?  I'm thinking about a hypothetical case when label A adds label B adds label A, etc. to infinity :)

Reply:
Just tested, if label A gets added but already existed on the issue, nothing changes/no new webhook gets sent